src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -134,37 +164,39 @@ private static (List<Error>, TableOfContentsInputModel) LoadTocModel(JToken tocT
             {
                 if (tocModelItem.Items != null && tocModelItem.Items.Any())
                 {
-                    errors.AddRange(ResolveTocModelItems(context, tocModelItem.Items, parents, filePath, rootPath, monikerComparer, monikerMap, resolveContent, resolveHref, resolveXref));
+                    errors.AddRange(ResolveTocModelItems(context, tocModelItem.Items, parents, filePath, rootPath, monikerMap, resolveContent, resolveHref, resolveXref));
                 }
 
                 var tocHref = GetTocHref(tocModelItem);
                 var topicHref = GetTopicHref(tocModelItem);
                 var topicUid = tocModelItem.Uid;
 
-                var (resolvedTocHref, resolvedTopicHrefFromTocHref, subChildren) = ProcessTocHref(tocHref);
+                var (resolvedTocHref, resolvedTopicItemFromTocHref, subChildren) = ProcessTocHref(tocHref);
                 var (resolvedTopicHref, resolvedTopicName, document) = ProcessTopicItem(topicUid, topicHref);
 
                 // set resolved href back
-                tocModelItem.Href = resolvedTopicHref ?? resolvedTopicHrefFromTocHref;
-                tocModelItem.TocHref = resolvedTocHref;
-                tocModelItem.Name = tocModelItem.Name ?? resolvedTopicName;
```

Comment:
please keep the assignment code unchanged, because the order is really import, just add moniker assignment code below

Reply:
suggest to have `GetMonikers` inner method to make the process more clear and clean

---
In reply to: [243163638](https://github.com/dotnet/docfx/pull/3797#discussion_r243163638) [](ancestors = 243163638)

Reply:
ok

Reply:
OK

---
In reply to: [243163726](https://github.com/dotnet/docfx/pull/3797#discussion_r243163726) [](ancestors = 243163726,243163638)